zospy/analyses/new/raysandspots/single_ray_trace.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+from typing import Literal
+from warnings import warn
+
+import pandas as pd
+from lark import Discard
+from pydantic import ConfigDict, Field, model_validator
+from pydantic.dataclasses import dataclass
+
+from zospy.analyses.new.base import AnalysisWrapper
+from zospy.analyses.new.parsers.transformers import ZospyTransformer
+from zospy.analyses.new.parsers.types import UnitField
+from zospy.api import constants
+
+
+class SingleRayTraceTransformer(ZospyTransformer):
+    def NAN(self, args):
```

Comment:
This does not seem to be used?

Reply:
`NAN` is defined in `single_ray_trace.lark`, so it is used. You can read how in Lark's documentation on transformers: https://lark-parser.readthedocs.io/en/stable/visitors.html#transformer
Summary: the methods of transformers must have the same name as the tokens defined in the grammar.